controllers/managedcluster.go

```diff
@@ -75,15 +87,16 @@ func (r *MultiClusterHubReconciler) ensureKlusterletAddonConfig(m *operatorsv1.M
 		return ctrl.Result{}, err
 	}
 
-	klusterletaddonconfig := getKlusterletAddonConfig()
+	newKlusterletaddonconfig := getKlusterletAddonConfig(m)
+	klusterletaddonconfig := newKlusterletaddonconfig.DeepCopy()
```

Comment:
Why is the copy needed? I think an empty instance would be sufficient for the `r.Client.Get` to work.

Reply:
empty need a kind but getKlusterletAddonConfig has everything I need

Reply:
That should be fine. There's not a lot of data that's copied since `newKlusterletaddonconfig` in minimal.